react/AppIcon/index.jsx

```diff
@@ -25,7 +25,28 @@ export class AppIcon extends Component {
   }
 
   componentDidMount() {
-    this.load()
+    const { status } = this.state
+    if (status !== 'DONE') {
+      this.load()
+    }
+  }
+
+  // When nested into a list of component, an AppIcon may update,
+  // and we must then update the state
+  componentDidUpdate(prevProps) {
+    const { app, domain, secure } = this.props
+
+    if (app.slug === prevProps.app.slug) return
+
+    const preloaded = getPreloaded(app, domain, secure)
+
+    if (preloaded) {
+      this.setState({
+        icon: preloaded
+      })
+    } else {
+      this.load()
+    }
```

Comment:
The management of icon/status/error is now done in 3 methods : `componentDidMount`, `componentDidUpdate` and the `constructor`. I think we should try to have less methods touching this shared state.

One problem I see here is that when the icon is set, the `error` is not reset. Also since `load()` can take time to set the state to the proper icon, the wrong icon can be shown for a while. I think the 3 methods should call something like `setIcon` that would be the only manager for error/icon and status.